driver-core/src/main/java/com/datastax/driver/core/DefaultResultSetFuture.java

```diff
@@ -156,7 +156,7 @@ public void run() {
                     setException(new DriverInternalError(String.format("Got unexpected %s response from %s", response.type, connection.address)));
                     break;
             }
-        } catch (RuntimeException e) {
+        } catch (Throwable e) {
```

Comment:
I think the change makes sense: if we don't catch a `Throwable` here, it goes to [Dispatcher#exceptionCaught](https://github.com/datastax/java-driver/blob/2.1/driver-core/src/main/java/com/datastax/driver/core/Connection.java#L1042) which will only log it at DEBUG level, then defuncts the connection. If the exception is thrown at the beginning of the method, the future is not completed and hangs forever.
By setting the future with the exception it's rethrown to the user code, which makes it much easier to debug.